src/utils.js

```diff
@@ -47,6 +47,53 @@ function removeLeadingNumbers (str) {
   return str.replace(/^[0-9|\.\-]+/, '');
 }
 
+
+/**
+ * Retrieve the correct parsing function for a file based on its
+ * path. Each parser with a `pattern` property will compile that pattern
+ * to a RegExp and test it against the filepath. If no match is found
+ * against any of the parsers by path pattern, a default parser will be
+ * returned: either a parser keyed by `default` in the `parsers` object
+ * or, lacking that, a default function that leaves the contents of the
+ * file untouched.
+ *
+ * @param {String} filepath
+ * @param {Object} parsers
+ * @see options module
+ * @return {Function} applicable parsing function for file contents
+ */
+function matchParser (filepath, parsers = {}) {
```

Comment:
Hah! I was this close to adding one. This PR is a little light on docs overall. Want to flesh out as I go. Noted.